src/main/java/duke/helpers/Parser.java

```diff
@@ -0,0 +1,117 @@
+                        throw new WrongBooleanException("Hey! \n" +
+                                "This is undone. You can't mark it undone again. :0 \n" +
+                                "try again.");
+                    }
+                    tasks.mark(splitArr[0], (Integer.parseInt(splitArr[1]) - 1));
+                } else if (splitArr[0].equals("delete")) {
+                    if ((Integer.parseInt(splitArr[1])) > tasks.getSize()) {
+                        throw new OutOfIndexException("Help! \n" +
+                                "The number has to be within range of our task-list!\n" +
+                                "Please try again!");
+                    }
+                    Task newTask = tasks.getTask(Integer.parseInt(splitArr[1]) - 1);
+                    tasks.removeTask(Integer.parseInt(splitArr[1]) - 1);
+                    System.out.println(Ui.formatStr(tasks.deleteReport(newTask)));
+                } else if (splitArr[0].equals("find")) {
+                    String searchTerm = input.substring(5);
+                    boolean hasFoundAnyTerms = false;
+                    String outputResults = "";
+                    for (int i = 0; i < tasks.getSize(); i++) {
+                        if (tasks.getTask(i).getContent().contains(searchTerm)) {
+                            if (outputResults == "") {
+                                outputResults += (i + 1) + tasks.getTask(i).toString();
+                                hasFoundAnyTerms = true;
+                            } else {
+                                outputResults += "\n" + (i + 1) + tasks.getTask(i).toString();
+                                hasFoundAnyTerms = true;
+                            }
+                        }
+                    }
+                    Ui.outputSearchResults(hasFoundAnyTerms, outputResults);
+                } else if (splitArr[0].equals("todo")) {
+                    Todo newTodo = new Todo(input);
+                    tasks.addTask(newTodo);
+                    System.out.println(Ui.formatStr(tasks.addReport(newTodo)));
+                    pw.write(newTodo.printRecord());
+                } else if (splitArr[0].equals("deadline")) {
+                    Deadline newDead = new Deadline(input);
+                    tasks.addTask(newDead);
+                    System.out.println(Ui.formatStr(tasks.addReport(newDead)));
+                    pw.write(newDead.printRecord());
+                } else if (splitArr[0].equals("event")) {
+                    Event newEvent = new Event(input);
+                    tasks.addTask(newEvent);
+                    System.out.println(Ui.formatStr(tasks.addReport(newEvent)));
+                    pw.write(newEvent.printRecord());
+                } else {
+                    throw new VagueInputException("Oh no! What do you mean? \n" +
+                            "I'm confused. Please specify... @.@");
+                }
+            } catch (VagueInputException ex) {
```

Comment:
I like how you the namings of the different Exceptions as I am able to understand clearly what possible errors handling inputs will result in!